.github/workflows/appetize.yml

```diff
@@ -19,144 +19,157 @@ on:
         required: false
 
 jobs:
+  prepare:
+    name: Preparing deploy
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.matrix.outputs.result }}
+    steps:
+      - name: 📋 Download SDK information
+        run: curl --silent -o versions.json https://exp.host/--/api/v2/versions
+      - name: 🔍 Resolve SDK versions
+        uses: actions/github-script@v4
+        if: github.event.inputs.sdkVersion
+        id: sdk
+        with:
+          script: |
+            const { sdkVersions } = require('./versions.json')
+            const version = '${{ github.event.inputs.sdkVersion }}'
+            const sdk = sdkVersions[version];
+
+            if (!sdk) throw new Error(`Could not find SDK "${version}"`)
+
+            core.info(`Resolved client versions for SDK "${version}"`)
+            core.info(`  - Android: ${sdk.androidClientVersion}`)
+            core.info(`  - iOS: ${sdk.iosClientVersion}`)
+
+            for (const key of ['androidClientVersion', 'androidClientUrl', 'iosClientUrl', 'iosClientVersion']) {
+              core.setOutput(key, sdk && sdk[key] ? sdk[key] : '')
+            }
+      - name: 👷‍♀️ Prepare tasks
+        uses: actions/github-script@v4
+        id: matrix
+        with:
+          script: |
+            const allQueues = ['main', 'embed']
+            const queue = '${{ github.event.inputs.appetizeQueue }}'
+
+            return {
+              include: allQueues
```

Comment:
This is a nice trick to create a matrix set for both `android` and `ios`. This matrix is based on the different Appetize queues that we have. E.g. for `main` **OR** `embed`, only 1 item is created. For `all`, both `main` and `embed` are added. This results in a workflow like this:

<img width="1029" alt="Screenshot 2021-08-11 at 21 50 28" src="https://user-images.githubusercontent.com/1203991/129093837-04429e48-4eaf-4cb4-ace3-f223ec2fb240.png">

Reply:
clever